File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/CustomNumberPicker.java
package com.example.sarahn.waterreminderapp.customsViews;

import android.content.Context;
import android.graphics.Color;
import android.graphics.Typeface;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.NumberPicker;
import android.widget.TextView;

import com.example.sarahn.waterreminderapp.classes.MyApplication;

import me.grantland.widget.AutofitHelper;

/**
 * Created by SarahN on 7/1/2017.
 */
public class CustomNumberPicker extends NumberPicker {
    public CustomNumberPicker(Context context) {
        super(context);
    }

    public CustomNumberPicker(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomNumberPicker(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void addView(View child) {
        super.addView(child);
        init(child);
    }

    @Override
    public void addView(View child, int index) {
        super.addView(child, index);
        init(child);

    }

    @Override
    public void addView(View child, int width, int height) {
        super.addView(child, width, height);
        init(child);

    }

    @Override
    public void addView(View child, ViewGroup.LayoutParams params) {
        super.addView(child, params);
        init(child);
    }

    @Override
    public void addView(View child, int index, ViewGroup.LayoutParams params) {
        super.addView(child, index, params);
        init(child);

    }

    private void init(View child){
        if(child instanceof TextView){
            Typeface face= Typeface.createFromAsset(MyApplication.getContext().getAssets(), "fonts/webcomic.ttf");
            ((EditText) child).setTextSize(13);
            ((EditText) child).setTextColor(Color.parseColor("#333333"));
            ((EditText) child).setTypeface(face);


        }
    }
}

File: app/src/main/java/com/example/sarahn/waterreminderapp/customsViews/CustomNumberPicker.kt
package com.example.sarahn.waterreminderapp.customsViews

import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.NumberPicker
import android.widget.TextView
import com.example.sarahn.waterreminderapp.classes.MyApplication

/**
 * Created by SarahN on 7/1/2017.
 */
class CustomNumberPicker : NumberPicker {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}

    override fun addView(child: View) {
        super.addView(child)
        init(child)
    }

    override fun addView(child: View, index: Int) {
        super.addView(child, index)
        init(child)
    }

    override fun addView(child: View, width: Int, height: Int) {
        super.addView(child, width, height)
        init(child)
    }

    override fun addView(child: View, params: ViewGroup.LayoutParams) {
        super.addView(child, params)
        init(child)
    }

    override fun addView(child: View, index: Int, params: ViewGroup.LayoutParams) {
        super.addView(child, index, params)
        init(child)
    }

    private fun init(child: View) {
        if (child is TextView) {
            val face = Typeface.createFromAsset(MyApplication.context?.getAssets(), "fonts/webcomic.ttf")
            (child as EditText).textSize = 13f
            child.setTextColor(Color.parseColor("#333333"))
            child.setTypeface(face)
        }
    }
}